.github/workflows/ci.yml

```diff
@@ -1,37 +1,61 @@
-name: nf-core CI
+name: test
 # This workflow runs the pipeline with the minimal test dataset to check that it completes without any syntax errors
 on:
-  push:
-    branches: [dev]
   pull_request:
   release:
     types: [published]
+  merge_group:
+    types:
+      - checks_requested
+    branches:
+      - master
+      - dev
 
 # Cancel if a newer run is started
 concurrency:
   group: ${{ github.workflow }}-${{ github.event.pull_request.number || github.ref }}
   cancel-in-progress: true
 
 jobs:
+  changes:
+    name: Check for changes
+    runs-on: ubuntu-latest
+    outputs:
+      # Expose matched filters as job 'tags' output variable
+      tags: ${{ steps.filter.outputs.changes }}
+    steps:
+      - uses: actions/checkout@v3
+      - uses: dorny/paths-filter@v2
+        id: filter
+        with:
+          filters: "tests/config/tags.yml"
+
   test:
-    name: Run pipeline with test data
-    # Only run on push if this is the nf-core dev branch (merged PRs)
-    if: "${{ github.event_name != 'push' || (github.event_name == 'push' && github.repository == 'nf-core/sarek') }}"
+    name: ${{ matrix.tags }} ${{ matrix.profile }} NF ${{ matrix.NXF_VER }}
     runs-on: ubuntu-latest
+    needs: changes
+    if: needs.changes.outputs.tags != '[]'
     strategy:
-      # HACK Remove after DSL2 rewrite is done
       fail-fast: false
       matrix:
+        tags: ["${{ fromJson(needs.changes.outputs.tags) }}"]
+        profile: ["docker", "singularity"]
```

Comment:
I noticed that some CI tests using Singularity failed. Can you explain why you run tests with both Docker and Singularity? Can you just use Docker?

Reply:
I think Maxime is looking into the failing Singularity-tests. As you may know, the ambition is for Sarek to be able to run in conda, docker and singularity, and so when possible (and when it doesn't take too many resources), we would also like to test the pipeline in all three environments. However, there is a lot of problems with conda (like, for instance, none of the sentieon stuff currently runs with conda), and so the conda test is often disabled. However, we definitely want both Docker tests and Singularity tests to pass.

Reply:
We have very limited resources on GHA unfortunately